modules/instrument_builder/jsx/react.questions.js

```diff
@@ -608,7 +608,7 @@ class AddElement extends Component {
     if (this.props !== undefined && this.props.element) {
       // Editing an element, set to elements state
       this.state = {
-        Options: Instrument.clone(this.props.element.Options),
+        Options: Instrument.clone(this.props.element.Options === undefined ? {} : this.props.element.Options ),
         Description: Instrument.clone(this.props.element.Description),
         Name: Instrument.clone(this.props.element.Name),
```

Comment:
Hi @kongtiaowang line number 613 needs to be changed to:
`Name: Instrument.clone(this.props.element.Name === undefined ? '' : this.props.element.Name),`
and it will fix the undefined error.

Reply:
I'm far from a javascript expert.. but wouldn't that be the same as (and more idiomatic as) `this.props.Element.Name || ''`?

Reply:
@driusan that works as well and can be used instead. 
I'm so fond of ? .. : ..; that I never think to write anything else.